✨ Change filename when no file is available

laurentsimon opened the pull request:
We currently display "on file available" when no file in the repo can be pointed to in the SARIF report. We use this, for example, for branch-Protection, Code-Review, Security-Policy. It's been reported to be confusing for users.
This PR changes the filename to be empty as " ". "" is not a valid name, which is why I provide a single space.

List of alerts:

![Screen Shot 2022-01-06 at 1 42 42 PM](https://user-images.githubusercontent.com/64505099/148456354-bc69e44c-e6c6-4391-8cf3-58de35164668.png)

When clicking on the alert, no name is shown as:

![Screen Shot 2022-01-06 at 1 45 24 PM](https://user-images.githubusercontent.com/64505099/148456536-d8c6257b-6fc7-4a25-8692-960ae7b02c01.png)

laurentsimon commented:
cc @olivekl

azeemshaikh38 reviewed (commented):
Do you mind adding a screenshot of what the updated UI would look like with this PR?

olivekl commented:
Agreed that "no file available" is confusing. I agree with using either a blank filename, or something like "not associated with any file."

laurentsimon commented:
> Do you mind adding a screenshot of what the updated UI would look like with this PR?

added the screenshot in the description. PTAL

laurentsimon commented:
we could also use as file the checks.md#check-name location... random idea

laurentsimon commented:
here's an example if I link to the documentation of checks:

List:
![Screen Shot 2022-01-06 at 6 43 13 PM](https://user-images.githubusercontent.com/64505099/148482912-53a252cf-0779-4f5a-8c18-2fc33ae20dba.png)

After clicking:

![Screen Shot 2022-01-06 at 6 44 07 PM](https://user-images.githubusercontent.com/64505099/148482985-e401a6c3-1228-4af4-b5e5-9c84126fb7c2.png)

laurentsimon commented:
friendly ping. 
Forget about the link to the `checks.md` suggestion, it does not work because the file does not live in the repo under analysis.
I've changed the file name to: `no file associated with this results`.
I've also added a `Click Remediation section below to solve this issue` below the highlighted text fr all the checks, and updated the text `Remediation:` to `Remediation (click "Show more" below):` (the GitHub UX requires an additional click that is not obvious). Thanks @olivekl for brainstorming this.

azeemshaikh38 commented:
I apologize, I'm caught up in getting the Scorecard site working and some other issues. Will take a look at this by EOD tomo.

azeemshaikh38 commented on pkg/sarif.go:
```diff
@@ -592,7 +590,8 @@ func (r *ScorecardResult) AsSARIF(showDetails bool, logLevel zapcore.Level,
 		RuleIndex := len(run.Tool.Driver.Rules) - 1
 		if len(locs) == 0 {
-			locs = addDefaultLocation(locs, "no file available")
+			// Use an "empty" filename to avoid confusing users.
```
Remove comment?

laurentsimon replied:
good catch. Will do!

azeemshaikh38 reviewed (approved):
LGTM
